src/variables.js

```diff
@@ -0,0 +1,266 @@
+import { useState, useEffect } from 'react';
+
+
+
+const getCSSVar = (variable) => {
+    return getComputedStyle(document.documentElement).getPropertyValue(variable).trim();
+  }
+
+// stores information for all posts
+const POSTS = [];
+
+const Users = [
+  {
+    name: 'First Last',
+    initials: 'FL',
+    color: getCSSVar('--purpleIcon'),
+    class: '202X',
+    school: ['A School'],
+    major: ['Some Interesting Major', 'Some Other Major'],
+    bio: 'Really interesting things about my background, interests, hobbies, etc.',
+    posts: [],
+    saved: []
+  },
+  {
+    name: 'John Doe',
+    initials: 'JD',
+    color: getCSSVar('--pinkIcon'),
+    class: '2025',
+    school: ['Bienen, Weinberg'],
+    major: ['Trumpet Performance', 'Psychology'],
+    bio: 'I like touching grass.',
+    posts: [],
+    saved: []
+  },
+  {
+    name: 'Some Person',
+    initials: 'SP',
+    color: getCSSVar('--yellowIcon'),
+    class: '2027',
+    school: ['McCormick'],
+    major: ['Mechanical Engineering'],
+    bio: 'I don\'t like touching grass.',
+    posts: [],
+    saved: []
+  }
+]
+
+const createPost = (user, location, start, end) => {
+  const post = {
+    name: user.name,
+    initials: user.initials,
+    iconColor: user.color,
+    location: location,
+    start: start,
+    end: end,
+    bookmarked: false,
+    show: true,
+  }
+
+  user.posts.push(post);
+  POSTS.push(post);
+} 
+
+// placeholders before create button functions are implemented
+createPost(Users[0], 'Sargent Dining Commons', new Date('2024-09-25T12:30'), new Date('2024-09-25T13:00'));
+createPost(Users[1], 'Foster Walker Plex East', new Date('2024-10-17T16:00'), new Date('2024-10-17T18:00'));
+createPost(Users[2], 'Allison Dining Commons/ Sargent Dining Commons', new Date('2024-10-14T20:00'), new Date('2024-10-14T21:30'));
+
+// current user
+const User = Users[0];
+
+// contains the displayed list of posts
+// profileList: true if list is the user's personal or saved lists
+// userSavedList: true if list is the user's saved list
+const PostList = ({Posts, profileList, userSavedList}) => {
+  const [posts, setPosts] = useState(Posts);
```

Comment:
A great use of useState!
When you want to change the current user to make posts on behalf of other users, you could add another useState for the current user.